LibSWBF2/InternalHelpers.cpp

```diff
@@ -37,38 +37,36 @@ namespace LibSWBF2
 		return Vector4(quaternion.x, quaternion.y, quaternion.z, quaternion.w);
 	}
 
+
 	List<uint32_t> TriangleStripToTriangleList(List<uint16_t> indexBuffer, uint32_t offset)
-	{
+	{		
 		List<uint32_t> result;
 
-		uint8_t triCount = 0;
-		size_t numIndices = indexBuffer.Size();
-		for (size_t i = 0; i < numIndices; ++i)
-		{
-			if (triCount == 3)
-			{
-				i -= 2;
-				triCount = 0;
-			}
-
-			result.Add(indexBuffer[i] + offset);
-			triCount++;
-		}
-
-		//after conversion to Triangle List, indices are listed CW CCW CW CCW, so let's flip them
-		bool flip = false;
-		size_t numTriangles = result.Size();
-		for (size_t i = 0; i < numTriangles; i += 3)
+		for (int i = 0; i < indexBuffer.Size() - 2; i++)
 		{
-			if (flip)
+			if (indexBuffer[i]   != indexBuffer[i+1] && //Check for degenerate tris
+				indexBuffer[i+1] != indexBuffer[i+2] && //
+				indexBuffer[i]   != indexBuffer[i+2])	//
 			{
-				std::swap(result[i], result[i + 2]);
+				if (i % 2 == 0) //swap clockwiseness 
+				{
+					result.Add(indexBuffer[i]   + offset);
+					result.Add(indexBuffer[i+1] + offset);
+					result.Add(indexBuffer[i+2] + offset);
+				} 
+				else 
+				{
+					result.Add(indexBuffer[i+1] + offset);
+					result.Add(indexBuffer[i]   + offset);
+					result.Add(indexBuffer[i+2] + offset);
+				}
 			}
```

Comment:
Changed this to skip degenerate triangles in the strip.  Tested with Unity model + terrain import.